docs/source/nlp/nemo_megatron/retro/retro_model.rst

```diff
@@ -1,469 +1,170 @@
-        model.tokenizer.merge_file=/dataset/gpt2-merges.txt \
-        model.tokenizer.vocab_file=/dataset/gpt2-vocab.json \
-        model.data.data_prefix=[/result/pubmed_eval_text_document] \
-        model.data.knn_index=[dataset/pubmed_knn_final.save] \
-        model.data.retrieval_prefix=/result/pubmed_eval_text_document \
-        model.micro_batch_size=8 \
-        model.shape_file=tp8_32depth_o1_rel_shape_info.yaml
-
-.. note:: We have chosen to use ``muadamw`` as the optimizer for use with the mu-transfer method.  Currently, only ``muadam`` and ``muadamw`` are supported. 
-
-Similarly to the pre-training in Option 1, the model nemo file can be found at the result checkpoint directory after training is complete.
-
-Run NeMo RETRO Model Inference
+        python /examples/nlp/language_modeling/megatron_retro_pretraining.py \
+            trainer.num_nodes=1 \
+            trainer.devices=8 \
+            trainer.precision=bf16 \
+            trainer.accelerator=gpu \
+            trainer.max_steps=750000
+            trainer.val_check_interval=10 \
+            trainer.precision=16 \
+            exp_manager.exp_dir=/path/to/exp_dir \
+            model.mcore_gpt=True \
+            model.tensor_model_parallel_size=1 \
+            model.pipeline_model_parallel_size=1 \
+            model.megatron_amp_O2=True \
+            model.retro.num_layers=12 \
+            model.retro.retro_encoder_num_layers=2 \
+            model.retro.retro_num_retrieved_chunks=2 \
+            model.retro.retro_project_dir=/path/to/retro_workdir \
+            model.micro_batch_size=4 \
+            model.data.num_workers=4 \
+            model.data.data_prefix=["none"] \
+            model.data.shuffle_documents=False \
+            model.data.dataloader_type=single \
+            model.data.splits_string=\'98,2,0\' \
+            model.optim.lr=6.0e-4 \
+            model.optim.weight_decay=0.1 \
+            model.optim.sched.name=CosineAnnealing \
+            model.optim.sched.min_lr=6.0e-5 \
+            model.optim.sched.max_steps=650000 \
+            model.optim.name=distributed_fused_adam
+
+During the training, we can monitor the process with Weights and Biases (WandB) by setting ``exp_manager.create_wandb_logger=True`` and set relevant wandb arguments.
+After training, the model distributed checkpoint directory can be found at the result checkpoint directory.
+
+Run RETRO Model Inference
 -------------------------------
 
-Once the NeMo RETRO model has been trained, we can put it into inference mode and experiment with it. 
-During inference, we are not limited to the static Faiss index that we built earlier for KNN queries. 
-We can feed any external data to the model as retrieval context. NeMo RETRO implementation supports dynamic retrieval service, 
-allowing users to add, reset, and query new documents on the fly.
+Once the RETRO model has been trained, we can put it into inference mode and experiment with it. 
```

Comment:
Once the RETRO model has been trained, you can put it into inference mode and experiment with it.